### ISSUE TITLE
Clarify solc unavailable message

### ISSUE THREAD

**Issue by carver** [ethereum/py-solc#40]
### What was wrong?

The error message is not clear, when solc isn't available.

Examples:

- https://ethereum.stackexchange.com/questions/26192/errno-2-no-such-file-or-directory-solc-using-py-solc-and-solidity
- https://github.com/ethereum/web3.py/issues/467

Proposal: catch and reraise the exception with a note saying something like:

> The command line tool `solc` wasn't found. Try `python -m solc.install v0.4.17` or see  http://solidity.readthedocs.io/en/latest/installing-solidity.html

#### Cute Animal Picture

![cute animal](http://cdn.mutually.com/wp-content/uploads/2017/07/mini-horse-filmed-sitting-in-a-high-chair-sparking-animal-rights-outragea-00_00_02_08-still013.jpg)

**Comment by jasonrhaas** [comment]
+1 for this.  Just ran into this in trying to get the web3.py quickstart to run.

**Comment by jasonrhaas** [comment]
Maybe adding a try/except here https://github.com/ethereum/py-solc/blob/master/solc/wrapper.py#L152-L155

is a good approach.  Like:

```
try:
    proc = subprocess.Popen(command,
                            stdin=subprocess.PIPE,
                            stdout=subprocess.PIPE,
                            stderr=subprocess.PIPE) 
except FileNotFoundError as e:
   raise FilenotFoundError("Added Error Message") from e
```

**Comment by pipermerriam** [comment]
I think the right place to catch this is in in [`solc/wrapper.py#L53`](https://github.com/ethereum/py-solc/blob/3a6de359dc31375df46418e6ffd7f45ab9567287/solc/wrapper.py#L53) and to use `solc.utils.filesystem.is_executable_available` to check that `solc_binary` is present.  We can then raise a `FileNotFound` error with a friendly error message.

**Comment by jasonrhaas** [comment]
@pipermerriam cool, I think I can probably add that in as part of the work I'm doing on Web3.py.  I ran into this as I was setting up a test bed for work on Web3.py.  Still having an issue actually even though I installed Solidity.  Must be a path or environment issue I'm guessing?

**Comment by pipermerriam** [comment]
Yeah, if `solc` isn't on your `$PATH` you can either add it to your `$PATH` or set the environment variable `SOLC_BINARY` to the path to the `solc` executable.

**Comment by grde1988** [comment]
Hello Gentlemen, 

Is there any clarity on this py-solc import issue? I have it installed but still not able to import the module into my python scripts: 

https://ethereum.stackexchange.com/questions/26192/errno-2-no-such-file-or-directory-solc-using-py-solc-and-solidity/40768#40768

`>>> import solc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named 'solc'`


Thank you for all the great work!

Ricardo

**Comment by carver** [comment]
@grde1988 that's a different issue (notice the difference between the errors: "No such file or directory" and "ImportError: No module named 'solc'"). Please ask a fresh stackexchange question, including your steps taken to set up, versions installed (using `pip freeze`), etc.

**Comment by AlexanderLife** [comment]
> FileNotFoundError: [Errno 2] No such file or directory: 'solc': 'solc'

short:
solc != solcjs
type solcjs --help
type solc --help
see the problem 
did you get  bash: solc: command not found for solc --help
if so type `sudo npm install -g solc-cli`

long bs:
it's not finding solc

so for me when you do a fresh `sudo npm install -g solc`  then when you type `solc --help` it says 

> bash: solc: command not found

 but I installed it. oh wait when I type  `solcjs --help` it works as expected guess we just had to guess 2 letters. 

I do `npm search solc` hummm solc-cli I bet cLi stands for Command Line Interface if I was a compiler I say something like undeclared identifier. this is something important to remember in documenting your code people are compilers, define don't just show 1337 scripts you could do with the cool tool. sorry, I'm a little frustrated with JavaScript people and their relation to solidity compile and deployment. hopefully abstracting to python will solve my problems. just saying everything with compile and deployment should be handled as preprocessor directives in solidity completely remove all JS and replace it with a real compiler. you could do something like #GAS: 2000000; #PORT: 9545; #LINKLIBRARY: <address of deployed library>;
I don't think it would really be that hard you could do something like python3 -m py-solc path/to/file.sol for compiling and deploying. oh how hypocritical of me I am not defining http://www.cplusplus.com/doc/tutorial/preprocessor/ , https://en.wikipedia.org/wiki/C_preprocessor.

anyway to fix this the above error we `sudo npm install -g solc-cli` after successful install we type `solc --help` it works as expected. 

hopefully the error has now changed to a easy one!

so another issue I had was I had two pythons 2 and 3 on my system and pip wanted to only install on python 2 the fix to that was to run pip as module `sudo python3 -m pip install py-solc` I am not really that experienced with python so it was a little hard for me but maybe it will help someone.
